### ISSUE TITLE
How to AnnotatedRegion?

### ISSUE THREAD

**Issue by badaween-ahmed** [BendixMa/sliding-sheet#7]
```dart
   await showSlidingBottomSheet(
      context,
      builder: (context) {
        return SlidingSheetDialog(
         //...
    
        );
      },
    );
```
Can't wrap AnnotatedRegion.

```dart
AnnotatedRegion<SystemUiOverlayStyle>(
    value: SystemUiOverlayStyle(
       systemNavigationBarIconBrightness:
         Theme.of(context).brightness == Brightness.dark
             ? Brightness.light
              : Brightness.dark,
            systemNavigationBarColor: Theme.of(context).backgroundColor,
          ),
        child: SlidingSheetDialog(
           //.. Can't do this
    ),
);
```

**Comment by BendixMa** [comment]
No, indeed with this is currently not possible. However when using `AnnotatedRegion` the `SystemUiOverlayStyle` wouldn't be set to the previous one when the route is being popped, which is I think what you want to achieve.

Instead you could try this:
~~~dart
SystemChrome.setSystemUIOverlayStyle(newStyle);

await showSlidingBottomSheet(
    context,
    builder: (context) {
      return SlidingSheetDialog(
        //...
      );
    },
  );

SystemChrome.setSystemUIOverlayStyle(previousStyle);
~~~

**Comment by badaween-ahmed** [comment]
Actually I have a bottom navigation bar only on the home screen, so when I open the sheet I want the navigation bar color to be the same as sheet color.  

![Screenshot_1582984009](https://user-images.githubusercontent.com/53362329/75608822-ffffe000-5b28-11ea-9e1d-3d71db89fa8a.png)

👇 Here I want the system navigation bar to be the same as background color(white).

![Screenshot_1582984614](https://user-images.githubusercontent.com/53362329/75608881-90d6bb80-5b29-11ea-9b27-16bc0891d473.png)

Previous AnnotatedRegion applied to change the system navigation bar to accent color (this blue color).

```dart
SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle(
      systemNavigationBarIconBrightness:
          Theme.of(context).brightness == Brightness.dark
              ? Brightness.light
              : Brightness.dark,
      systemNavigationBarColor: Theme.of(context).backgroundColor,
    ));
```
This solution didn't work system navigationbar changed to background for once in the begining then it again change to the accent color (this blue color).

**Comment by BendixMa** [comment]
I released a new version `0.2.7` which has a new parameter `parentBuilder` which allows you to wrap the sheet inside a parent.

So you can now do something like this:

~~~dart
await showSlidingBottomSheet(
    context,
    parentBuilder: (context, sheet) {
      return AnnotatedRegion<SystemUiOverlayStyle>(
        value: SystemUiOverlayStyle(
          systemNavigationBarIconBrightness:
              Theme.of(context).brightness == Brightness.dark ? Brightness.light : Brightness.dark,
          systemNavigationBarColor: Theme.of(context).backgroundColor,
        ),
        child: sheet,
      );
    },
    builder: (context) {
      return SlidingSheetDialog(
        //...
      );
    },
  );
~~~

**Comment by badaween-ahmed** [comment]
It worked. Thanks for your great support.
